### PR TITLE
Updated to pyo3 and changed to new workflows

### PR DESCRIPTION
**Pull request by kbarkhqs**

### CODE REVIEW

File: .github/workflows/hqs-build-deploy-rust-pyo3.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# Build linux wheels and upload them to PyPi
+  build_maturin_builds_linux:
+    name: maturin_build-linux
+    needs: deploy_cratesio
+    uses: HQSquantumsimulations/reusable_workflows/.github/workflows/reusable_build_maturin_linux.yml@main
+    with: 
+      py_interface_folder: "struqture-py"
+      deploy: true
+      # Whether to build for aarch64. Can fail if there are C/C++ dependencies
+      build_for_arm: true
```
Comment (mlodi-hqs) [inline .github/workflows/hqs-build-deploy-rust-pyo3.yml:37]: The default option on the original workflow is false, is it ok to set it to true?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 This changelog track changes to the struqture project starting at version v1.0.0
 
+## 1.0.1
+
+* Updated to pyo3 0.18 and test-case 3.0
+* Updated to the new workflows
```
Comment (mlodi-hqs) [inline CHANGELOG.md:8]: Nico said that since workflows are not technically part of the package, they should not be mentioned here

**Review by mlodi-hqs — CHANGES_REQUESTED** [review]
- I'm pretty sure maturin can be upped to >=0.14,<0.15 without any problem
- The DEPENDENCIES file needs an update
- introduction.md still mentions `1.0.0`. (Is there a way to automatize that? Like making it read the current version. It's always very easy to forget)

File: .github/workflows/hqs-build-deploy-rust-pyo3.yml (hunk 1 above)
Comment (kbarkhqs) [reply .github/workflows/hqs-build-deploy-rust-pyo3.yml:37]: There aren't any C++ dependencies, I think it should be fine

File: CHANGELOG.md (hunk 2 above)
Comment (kbarkhqs) [reply CHANGELOG.md:8]: Thank you!

**Review by nfwvogt — APPROVED** [review]

File: .github/workflows/hqs-ci-test-rust-pyo3.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      windows: true
+      # Run tests also on macos runners
+      macos: true
+      rust_package_name: "struqture"
+
+  build_tests:
+    uses: HQSquantumsimulations/reusable_workflows/.github/workflows/reusable_build_tests_rust_pyo3.yml@main
+    with: 
+      # Run tests also on windows runners
+      windows: false
```
[the remark this answers is not in the document]
Comment (kbarkhqs) [reply .github/workflows/hqs-ci-test-rust-pyo3.yml:25]: Changed to true

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kbarkhqs) [reply CHANGELOG.md:8]: Removed

**Pull request merged**